### PR TITLE
fix transformation of decorators

### LINKED ISSUE #33: Babel parse errors for decorators
I tried to add `@babel/plugin-proposal-decorators` as I thought that it might fix this parsing issue. Do you have an idea how to fix this?

Screen Shot 2021-07-19 at 2 25 05 pm

### PR DESCRIPTION
**Pull request by meandmax**
fixes https://github.com/zxbodya/flowts/issues/33

### CODE REVIEW

File: packages/babel-plugin-flow-to-typescript/test/visitors/ClassDeclaration.test.ts [hunk 1]
```diff
@@ -56,6 +56,27 @@ test('Extending class declaration types are transformed', () => {
...
+test('Classes with decorators are transformed', () => {
+  const result = testTransform(
+    `
+    @injectIntel
+    export class C extends React.Component<?string, ?(string | boolean)> {}
+    `
+  );
+  expect(result.babel).toMatchInlineSnapshot(
+    `
+    "export @injectIntel
```
Comment (meandmax) [inline packages/babel-plugin-flow-to-typescript/test/visitors/ClassDeclaration.test.ts]: @zxbodya This does not seem to be correct, right. What else needs to be modified to make this work?

Comment (meyer) [reply packages/babel-plugin-flow-to-typescript/test/visitors/ClassDeclaration.test.ts]: [it’s configurable i think](https://babeljs.io/docs/en/babel-plugin-proposal-decorators#decoratorsbeforeexport). might need to be detected and conditionally enabled/disabled?

Comment (zxbodya) [reply packages/babel-plugin-flow-to-typescript/test/visitors/ClassDeclaration.test.ts]: indeed - config option for babel generator works - pushed a commit here

also it looks typescript supports only option with decorators before export, so I guess it make sense to add that config

File: packages/babel-plugin-flow-to-typescript/src/index.ts [hunk 2]
```diff
@@ -81,6 +81,7 @@ export default (_babel: any, opts: PluginOptions = {} as PluginOptions) => {
       parserOpts.plugins.push('objectRestSpread');
       parserOpts.plugins.push('optionalChaining');
       parserOpts.plugins.push('nullishCoalescingOperator');
+      parserOpts.plugins.push('decorators-legacy');
```
Comment (zxbodya) [inline packages/babel-plugin-flow-to-typescript/src/index.ts]: I am not very familiar with using decorators, what is the difference between this and `decorators` plugin?

as an option this probably can be: 
```js
parserOpts.plugins.push([
    'decorators',
    {
      decoratorsBeforeExport: true,
    },
  ]);
```
but I am not sure what would be actual difference…

Comment (zxbodya) [reply packages/babel-plugin-flow-to-typescript/src/index.ts]: if, I understand correctly:
- `decorators-legacy` - is stage 1 proposal version
- `decorators` - current, stage 2 proposal version, which looks to be available to use in TypeScript

I guess, we should use a new one…

Comment (meandmax) [reply packages/babel-plugin-flow-to-typescript/src/index.ts]: I tried both and both worked but was not sure which one to use. I pushed a commit.

**Review by zxbodya — APPROVED** [review]

**Pull request merged**